### ISSUE TITLE
"Unauthorized: Invalid tunnel secret" error when trying to run a tunnel

### ISSUE THREAD

**Issue by bassamanator** [cloudflare/cloudflared#519]
I thought I had solved this issue but after a reboot, it's back.

I have 2 tunnels that were running fine. After rebooting, one of the tunnels runs, however, when I try to run the second one, I get the following error that repeats and then the `Tunnel server stopped`:
>2021-12-01T22:40:46Z INF Initial protocol http2
2021-12-01T22:40:46Z INF Starting metrics server on 127.0.0.1:36761/metrics
2021-12-01T22:40:46Z ERR Register tunnel error from server side error="Unauthorized: Invalid tunnel secret" connIndex=0

I tried deleting the problem tunnel, deleting the DNS record from my dashboard, created a new tunnel, edited my config.yml, and I still can't start the tunnel.

What am I doing wrong?

Thanks!

_Originally posted by @bassamanator in https://github.com/cloudflare/cloudflared/issues/426#issuecomment-982814984_

**Comment by sudarshan-reddy** [comment]
Do you know where your credentials file is picked from? You can force this to be explicit by using `--credentials-file </path/to/your/credentials/file>` when you run your tunnel.

**Comment by bassamanator** [comment]
@sudarshan-reddy 

I tried that and it didn't help. (--credentials-file...)

I'm following the [official guide](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/tunnel-guide) (which works for 1 of my tunnels without fail).

1. cloudflared tunnel create tunnel1-name
2. Create config.yml
> url: http://localhost:8000
tunnel: <Tunnel-UUID>
credentials-file: /home/bassam/.cloudflared/<Tunnel-UUID>.json
3. cloudflared tunnel route dns tunne1-name tunnel1
4. cloudflared tunnel --config /home/bassam/config.yaml run

What am I doing wrong?

**Comment by sudarshan-reddy** [comment]
This just means that the credentials file you are using isn't the right one. You might want to open it and see if it has the same tunnel uuid as your trying to run.

**Comment by bassamanator** [comment]
@sudarshan-reddy 

It is correct. I have verified this.

**Comment by bassamanator** [comment]
Here's my strapi.yml
># strapi.
url: http://localhost:1337
tunnel: AAABBB
credentials-file: /home/bassam/.cloudflared/AAABBB.json

AAABBB.json
>{
	"AccountTag": "generated for me",
	"TunnelSecret": "other generated for me",
	"TunnelID": "AAABBB",
	"TunnelName": "strapi-tunnel"
}

**Comment by bassamanator** [comment]
Is it okay to have multiple tunnels list in the config.yml?
>url: http://localhost:8000
tunnel: tunnel-uuid1
credentials-file: /root/.cloudflared/tunnel-uuid1.json

>url: http://localhost:8001
tunnel: tunnel-uuid2
credentials-file: /root/.cloudflared/tunnel-uuid2.json

I removed the second tunnel from the config and it still has an issue :/

**Comment by kenzik** [comment]
I've ran into this issue, too. It happens sometimes with > 1 tunnels in config.yml.

OS: OSX 12.1 - M1
Version: cloudflared version 2022.1.2 (built 2022-01-13-1258 UTC) 
Command: cloudflared tunnel <UID>

I encountered the issue when trying to run a third tunnel. Two of the tunnels would run fine -- any 2 of the 3 from the config. But the third would error with the invalid secret message.

My initial troubleshooting was to comment out the tunnel configs in config.yml for the 2 running tunnels and then start the 3rd tunnel. That worked every time with any permutation.

A less annoying way is to just use a separate config file for any additional tunnels, which is working for me.

Since you were still having trouble after removing your second tunnel, then YMMW of course -- just wanted to note this here for the devs.
